src/pi/SmartWebSocket.ts

```diff
@@ -0,0 +1,57 @@
+import WebSocket from './WebSocket';
+
+export interface Message {
+  [index: string]: any;
+}
+
+function awaitWebsocketReady(websocket: WebSocket) {
+  return new Promise((resolve, reject) => {
+    if (websocket.readyState === WebSocket.OPEN) {
+      resolve();
+      return;
+    }
+
+    websocket.addEventListener('open', () => {
+      resolve();
+    });
```

Comment:
Should this not reject at some point if the connection fails? It looks like never resolves if an error occurs.